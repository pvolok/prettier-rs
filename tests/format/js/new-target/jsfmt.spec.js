run_spec(import.meta, ["babel", "flow", "typescript"], {
  errors: {
    acorn: true,
    espree: true,
    meriyah: true,
    flow: true,
  },
});
