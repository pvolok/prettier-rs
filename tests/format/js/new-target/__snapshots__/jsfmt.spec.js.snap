// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outside-functions.js [acorn] format 1`] = `
"'new.target' can only be used in functions and class static block (1:11)
> 1 | const x = new.target;
    |           ^
  2 |"
`;

exports[`outside-functions.js [espree] format 1`] = `
"'new.target' can only be used in functions and class static block (1:11)
> 1 | const x = new.target;
    |           ^
  2 |"
`;

exports[`outside-functions.js [flow] format 1`] = `
"Unexpected token \`.\` (1:14)
> 1 | const x = new.target;
    |              ^
  2 |"
`;

exports[`outside-functions.js [meriyah] format 1`] = `
"new.target only allowed within functions (1:20)
> 1 | const x = new.target;
    |                    ^
  2 |"
`;

exports[`outside-functions.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
const x = new.target;

=====================================output=====================================
const x = new.target;

================================================================================
`;

exports[`range.js [acorn] format 1`] = `
"'new.target' can only be used in functions and class static block (2:1)
  1 | const x =
> 2 | new.target;
    | ^"
`;

exports[`range.js [espree] format 1`] = `
"'new.target' can only be used in functions and class static block (2:1)
  1 | const x =
> 2 | new.target;
    | ^"
`;

exports[`range.js [flow] format 1`] = `
"Unexpected token \`.\` (2:4)
  1 | const x =
> 2 | new.target;
    |    ^"
`;

exports[`range.js [meriyah] format 1`] = `
"new.target only allowed within functions (2:10)
  1 | const x =
> 2 | new.target;
    |          ^"
`;

exports[`range.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
rangeEnd: 52
rangeStart: 38
     |                                                                                | printWidth
=====================================input======================================
  1 | class A {
  2 |   constructor() {
> 3 |     const x =
    |           ^^^
> 4 | new.target;
    | ^^^^^^^^^^
  5 |   }
  6 | }
  7 |
=====================================output=====================================
class A {
  constructor() {
    const x = new.target;
  }
}

================================================================================
`;
